src/socket-to-conn.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const abortable = require('abortable-iterator')
+const { CLOSE_TIMEOUT } = require('./constants')
+const toMultiaddr = require('libp2p-utils/src/ip-port-to-multiaddr')
+
+const log = require('debug')('libp2p:websockets:socket')
+
+// Convert a stream into a MultiaddrConnection
+// https://github.com/libp2p/interface-transport#multiaddrconnection
+module.exports = (socket, options = {}) => {
+  const maConn = {
+    async sink (source) {
+      if (options.signal) {
+        source = abortable(source, options.signal)
+      }
+
+      try {
+        await socket.sink(source)
+      } catch (err) {
+        if (err.type !== 'aborted') {
+          log(err)
+        }
+      }
+    },
+
+    source: options.signal ? abortable(socket.source, options.signal) : socket.source,
+
+    conn: socket,
+
+    localAddr: undefined,
+
+    // If the remote address was passed, use it - it may have the peer ID encapsulated
+    remoteAddr: options.remoteAddr || toMultiaddr(socket.remoteAddress, socket.remotePort),
+
+    timeline: { open: Date.now() },
+
+    close () {
+      return new Promise(async (resolve) => { // eslint-disable-line no-async-promise-executor
```

Comment:
I think we should change the usage here so we're not using async inside the promise constructor.

We could use [p-timeout](https://github.com/sindresorhus/p-timeout), pass it `socket.close()` and await that.

Reply:
Cool, I found a bug thanks to this on my `it-ws` branch. Already updating it and changing this in favor of `p-timeout`